client/src/common/client.ts

```diff
@@ -317,6 +317,15 @@ interface _Middleware {
 	window?: WindowMiddleware;
 }
 
+// A general middleware is applied to both requests and notifications
+interface GeneralMiddleware {
+	provideRequestOrNotification?<R>(
+		this: void,
+		sendRequestOrNotification: (type: string | MessageSignature, ...params: any[]) => Promise<R>,
```

Comment:
We should also follow the style of the other middleware and make the next handler the last argument and call it next

Reply:
I did rename it to `next`, however I kept it second to last to be able to continue to use the rest parameters `...params`.  I made an attempt to not use a rest parameter, but ended up hitting [this error in connection](https://github.com/microsoft/vscode-languageserver-node/blob/84b4af76874ef32a0edb62e33dd2ad9a700c33ea/jsonrpc/src/common/connection.ts#L1230).

Let me know if you have an idea on how to restructure so that the rest parameter is not needed.